tools/golang-json-rpc-tests/main.go

```diff
@@ -0,0 +1,412 @@
+    testEstimateGas(client, fromAddress)
+    testGetGasPrice(client)
+    testGetAccounts(client)
+    testBlockByHash(client, blockHash)
+    testFeeHistory(client, 5, newestBlock, []float64{10, 50, 90})
+    testGetBlockTransactionCountByHash(client, blockHash)
+    testGetBlockTransactionCountByNumber(client)
+    testCodeAt(client, fromAddress)
+    testGetLogs(client, fromAddress, nil)
+    testStorageAt(client, fromAddress, "0x0")
+    testGetTransactionByBlockHashAndIndex(client, blockHash, txIndex)
+    testGetTransactionByBlockNumberAndIndex(client, blockHash, txIndex)
+    testGetTransactionByHash(client, signedTx.Hash().Hex())
+    testGetTransactionCount(client)
+    testGetTransactionReceipt(client, signedTx.Hash().Hex())
+    testSyncing(client)
+}
+
+func testBlockByHash(client *ethclient.Client, blockHash common.Hash) {
+    block, err := client.BlockByHash(context.Background(), blockHash)
+    if err != nil {
+        log.Fatalf("Failed to get block by hash: %v", err)
+    }
+    fmt.Printf("Block by hash: %s\n", block.Hash().Hex())
+}
+
+func testBlockByNumber(client *ethclient.Client, blockNumber *big.Int) {
+    block, err := client.BlockByNumber(context.Background(), blockNumber)
+    if err != nil {
+        log.Fatalf("Failed to get block by number: %v", err)
+    }
+    if block.Number().Cmp(blockNumber) != 0 {
+        log.Fatalf("Block number mismatch: expected %s, got %s", blockNumber.String(), block.Number().String())
+    }
+    fmt.Printf("Block by number\n")
+}
+
+func testTransactionByHash(client *ethclient.Client, txHash string) {
+    tx, isPending, err := client.TransactionByHash(context.Background(), common.HexToHash(txHash))
+    if err != nil {
+        log.Fatalf("Failed to get transaction by hash: %v", err)
+    }
+    fmt.Printf("Transaction by hash: %s (Pending: %t)\n", tx.Hash().Hex(), isPending)
+}
+
+func testTransactionReceipt(client *ethclient.Client, txHash string) {
+    receipt, err := client.TransactionReceipt(context.Background(), common.HexToHash(txHash))
+    if err != nil {
+        log.Fatalf("Failed to get transaction receipt: %v", err)
+    }
+    if receipt == nil {
+        fmt.Printf("Transaction receipt: null\n")
+    } else if receipt.TxHash.Hex() != txHash {
+        log.Fatalf("Receipt transaction hash mismatch: expected %s, got %s", txHash, receipt.TxHash.Hex())
+    } else {
+        fmt.Printf("Transaction receipt: %d\n", receipt.Status)
+    }
+}
+
+func testGetBalance(client *ethclient.Client, fromAddress common.Address) {
+    balance, err := client.BalanceAt(context.Background(), fromAddress, nil)
+    if err != nil {
+        log.Fatalf("Failed to get balance: %v", err)
+    }
+    if balance == nil {
+        log.Fatalf("Balance is nil")
+    }
+    fmt.Printf("Account balance: %s\n", balance.String())
+}
+
+func testEthCall(client *ethclient.Client, fromAddress common.Address) {
+    msg := ethereum.CallMsg{
+        From: fromAddress,
+        To:   &fromAddress,
```

Comment:
Should we try a real SC's address to make sure it works in practice? Otherwise, this will return an empty array.

Reply:
Plus I believe the program is missing a test case for `eth_sendRawTransaction`. We can use this test case to deploy an SC. This way, we can have an SC to work with and adequately cover the functionality of `eth_sendRawTransaction`.

Reply:
@quiet-node 

Go SC deployment and interactions tests were already covered in this PR: 
https://github.com/hashgraph/hedera-json-rpc-relay/pull/2584 .

But thanks for your feedback. I guess I'll add sample SC to this tests as well.

Reply:
@quiet-node 

Go SC deployment and interactions tests were already covered in this PR: 
https://github.com/hashgraph/hedera-json-rpc-relay/pull/2584 .

But thanks for your feedback. I guess I'll add sample SC to this tests as well.

Reply:
Thanks it would be nice!

Reply:
As you proposed, I’ve added smart contract deployment to the tests. This smart contract emits an event and has a storage slot filled, so there is an actual result when collecting logs or investigating the storage of this contract.

Reply:
@quiet-node As you proposed, I’ve added smart contract deployment to the tests. This smart contract emits an event and has a storage slot filled, so there is an actual result when collecting logs or investigating the storage of this contract.